### PR TITLE
Attempt to fix Windows build failures on CI

### PR DESCRIPTION
**Pull request by keflavich**
The problems are all invocation errors:
https://github.com/radio-astro-tools/spectral-cube/runs/4475860538?check_suite_focus=true#step:5:255

but it seems to be pointing at a file not closed error:
https://github.com/radio-astro-tools/spectral-cube/runs/4475860538?check_suite_focus=true#step:5:153
```
..\..\spectral_cube\spectral_axis.py .                                   [  0%]
..\..\spectral_cube\tests\test_analysis_functions.py ................... [  1%]
                                                                         [  1%]
..\..\spectral_cube\tests\test_casafuncs.py ......ssssssssssssss         [  2%]
..\..\spectral_cube\tests\test_cube_utils.py .......                     [  2%]
..\..\spectral_cube\tests\test_dask.py ....s...E                         [  3%]
..\..\spectral_cube\tests\test_io.py .E.E.E.E.E.E.E.E.E.E.E.E.E.E.E.E.E.E [  4%]
```
so... let's see if we can fix it.

### CODE REVIEW

**Comment by keflavich** [comment]
There was a bug introduced in pvextractor that caused gui stuff to fail

**Comment by keflavich** [comment]
Unfortunately I've uncovered a real error (a regression) caused by some change in region handling.  Something about compound region bboxes has changed, breaking our tests

**Comment by keflavich** [comment]
Some things were really broken, including one of our tests.

Both of these tests were incorrectly only getting 1 pixel along the spectral direction, but investigating in ds9 showed that 2 should've been included:
```
                             (('fk5.reg', (slice(None), 1, slice(None))),
                              ('fk5_twoboxes.reg', (slice(None), 1, slice(None))),
```
image

**Comment by keflavich** [comment]
@astrofrog @e-koch I'm requesting reviews from both of you because I made some significant logical changes in the region handling.  I think we had an uncaught error (actually an error in our tests) previously.  It would be good to have an independent look at that if possible

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -49,7 +49,7 @@ jobs:
           - os: windows-latest
             python-version: 3.9
             name: Python 3.9, all dependencies, and dev versions of key dependencies on Windows
-            toxenv: py39-test-all-dev
+            toxenv: py39-test-all-dev-noopenfiles
```
Comment (e-koch) [inline .github/workflows/main.yml:52]: Is the pvextractor gui still causing the open file issue? If not, run one with and without?

**Review by e-koch — CHANGES_REQUESTED** [review]
The changes look good to me. `get_overlap_slices` from the whole cube size is definitely a key fix, I think.

Two changes to add below, since they're not within the current "diff" region:
* Pin `regions>=0.3` (or higher??) in `all`: https://github.com/radio-astro-tools/spectral-cube/blob/master/setup.cfg#L47.
* Let's add a shape/size check to `test_ds9region_new` to cover checking output slice and data values here: https://github.com/radio-astro-tools/spectral-cube/blob/ccf301f50d036a853661c4a15c11f7b6a2e7084e/spectral_cube/tests/test_subcubes.py#L142
```
        assert sc.size == data[result].size
        assert sc.filled_data[:].squeeze().shape == data[result].shape

```

**Comment by keflavich** [comment]
@e-koch you want to make those suggested changes directly on this PR?

**Comment by e-koch** [comment]
@keflavich Done!

**Review by e-koch — APPROVED** [review]

**Pull request merged**